spyder/widgets/sourcecode/utils/decoration.py

```diff
@@ -50,6 +49,21 @@ def remove(self, decoration):
         except ValueError:
             return False
 
+    def extend(self, decorations):
```

Comment:
I still prefer add :-p

Reply:
`extend` is different to `add`. This function extends a list with another list
Maybe `append` could be changed to `add`

Reply:
Well sounds like extend and append for this hight level case should be handled as a single thing/method/function. I know this was supposed to look like a list but it makes no sense that it has to follow the list primitives, this is a higher level object, not a list.

Reply:
Ok, then I'll fusion append and extend into a function named add